string frame_of_reference
---
string tui_state